contracts/p1/mixins/RecollateralizationLib.sol

```diff
@@ -391,9 +414,36 @@ library RecollateralizationLibP1 {
             }
         }
 
+        // Special-case RToken
+        {
+            RTokenAsset rTokenAsset = RTokenAsset(
+                address(ctx.ar.toAsset(IERC20(address(ctx.rToken))))
+            );
+            uint192 bal = rTokenAsset.bal(address(ctx.bm));
+
+            if (
+                ntpCtx.rTokenPrice.high > 0 &&
+                isBetterSurplus(
+                    maxes,
+                    CollateralStatus.SOUND,
```

Comment:
why are we making the assumption that the rtoken is sound? it matches the previous logic but im just wondering if that is correct here

Reply:
The `RTokenAsset' isn't a collateral, it's just an asset. In the loop above you can see that all the other assets are also written to essentially have a SOUND collateral status for purposes of comparison.

Also, at the top of `manageTokens()` there's a require that the basket is SOUND.